devicetypes/vision-raytseng/vision-4-in-1-motion-sensor.src/vision-4-in-1-motion-sensor.groovy

```diff
@@ -0,0 +1,510 @@
+	parameter.parameter6type = "enum"
+	parameter.parameter6options = ["Highest", "Higher", "High", "Medium", "Low", "Lower", "Lowest"]
+	parameter.parameter6enumMap = ["Highest": 1, "Higher": 2, "High": 3, "Medium": 4, "Low": 5, "Lower": 6, "Lowest": 7]
+	parameter.parameter6default = "Medium"
+	parameter.parameter6size = 1
+	
+	parameter.parameter7name = "LedDispMode"
+	parameter.parameter7title = "LED display mode"
+	parameter.parameter7description = ""
+	parameter.parameter7type = "enum"
+	parameter.parameter7options = ["LED off when Temperature report/Motion active",
+								   "LED blink when Temperature report/Motion active",
+								   "LED blink when Motion active/LED off when Temperature report"]
+	parameter.parameter7enumMap = ["LED off when Temperature report/Motion active": 1,
+								   "LED blink when Temperature report/Motion active": 2,
+								   "LED blink when Motion active/LED off when Temperature report": 3]
+	parameter.parameter7default = "LED off when Temperature report/Motion active"
+	parameter.parameter7size = 1
+	
+	parameter.parameter8name = "RetryTimes"
+	parameter.parameter8title = "Motion notification retry times"
+	parameter.parameter8description = ""
+	parameter.parameter8type = "number"
+	parameter.parameter8options = "0..10"
+	parameter.parameter8enumMap = []
+	parameter.parameter8default = 3
+	parameter.parameter8size = 1
+	
+	return parameter."parameter${num}${text}"
+}
+
+def wakeUpIntervalUpdate(interval) {
+	def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+	
+	if (value != null) {
+		if (state."${getWakeUpInfo("name")}" != value) {
+			state."${getWakeUpInfo("name")}" = value
+			state."${getWakeUpInfo("name")}Refresh" = true
+		}
+	}
+}
+
+def wakeUpIntervalCheck(interval) {
+	def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+	
+	if (value != null) {
+		if (state."${getWakeUpInfo("name")}" == value) {
+			if (state."${getWakeUpInfo("name")}Refresh" == true) {		
+				state."${getWakeUpInfo("name")}Refresh" = false
+			}
+		}
+	}
+}
+
+def getWakeUpInfo(text) {
+	def wakeUp = [:]
+	
+	wakeUp.name = "wakeUpInterval"
+	wakeUp.title = "Wake up interval [Hours]"
+	wakeUp.description = ""
+	wakeUp.type = "number"
+	wakeUp.range = "1..4660"
+	wakeUp.default = 24
+	
+	return wakeUp."${text}"
+}
+
+private getCommandClassVersions() {
+	[
+		0x5E: 2,
+		0x22: 1,
+		0x85: 2,
+		0x59: 3,
+		0x80: 1,
+		0x70: 2,
+		0x5A: 1,
+		0x7A: 5,
+		0x87: 3,
+		0x72: 2,
+		0x8E: 3,
+		0X31: 5,
+		0x71: 3,
+		0x73: 1,
+		0x98: 1,
+		0x9F: 1,
+		0x6C: 1,
+		0x55: 2,
+		0x86: 3,
+		0x84: 2 
+	]
+}
+
+def parse(String description) {
+	def result = []
+	def cmd = zwave.parse(description, commandClassVersions)
+	
+	if (cmd) {
+		result += zwaveEvent(cmd)
+	}
+	else {
+		logDebug "Unable to parse description: ${description}"
+	}
+	
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	def cmds = []
+	
+	cmds += configure()
```

Comment:
This is battery-powered device that only wakes up and communicates when an event has occurred. 
If the smartThings hub would like to communicate with this device, the smartThings hub must wait for this device awake. When this device wakes up, it will send WakeUpNotification to inform the smartThings hub it is ready to communicate, and then the smartThings hub could send the configuration from the user settings to this device.

Reply:
That way configure() will be called regardless changes in settings and every few hours the device will receive the set of commands even if user doesn't intent to configure the device. You can improve that approach by using another state variables to determine if the configuration has to be done or not (for example: "Z-Wave Motion Sensor")

Reply:
Actually, the device does not receive the set of commands every few hours if the settings have not changed, since the configure() that has check which configuration is changed by the user and then send to the device. You could check the code of updated() and configure(). In updated(), it will set the corresponding flag if user changes in settings, and then in configure() it will check the flag and decide which command needs to be sent to the device .